pkg/controller/cloud/pvlcontroller.go

```diff
@@ -196,14 +190,14 @@ func (pvlc *PersistentVolumeLabelController) addLabelsToVolume(vol *v1.Persisten
 	// Only add labels if in the list of initializers
 	if needsInitialization(vol.Initializers, initializerName) {
 		if vol.Spec.AWSElasticBlockStore != nil {
```

Comment:
The checks for the AWSElasticBlockStore and the GCEPersistentDisk shouldn't be needed anymore, right?  The controller should query the configured cloud to get the labels for a given pv, and it is up to that cloud provider impl to provide the labels for the volume.  That would allow you to compress this to a single try to get labels.

Reply:
Ack, let me try to do that. Was trying to avoid making too many changes for 1.8

Reply:
yeah, as commented in https://github.com/kubernetes/kubernetes/issues/51629#issuecomment-326703532, something like this should be sufficient and easier
```
volumeLabels, err := pvlc.GetLabelsForVolume(vol)
if err != nil {
	return fmt.Errorf("error querying cloud provider for volume labels: %v", err)
}
```

Reply:
Done